phoenicis-core/src/main/java/com/playonlinux/core/webservice/HTTPDownloader.java

```diff
@@ -27,106 +27,99 @@
-import com.playonlinux.core.observer.ObservableDefaultImplementation;
+import com.playonlinux.core.entities.ProgressEntity;
+import com.playonlinux.core.entities.ProgressState;
 
-public class HTTPDownloader extends ObservableDefaultImplementation<ProgressStateEntity> {
+public class HTTPDownloader {
     private static final String EXCEPTION_ITEM_DOWNLOAD_FAILED = "Download of %s has failed";
 
     private static final int BLOCK_SIZE = 1024;
     private final URL url;
-    private State state;
     private float percentage;
-
-    public enum State {
-	READY, PROGRESSING, SUCCESS, FAILED
-    }
+    private Consumer<ProgressEntity> onChange;
 
     public HTTPDownloader(URL url) {
-	this.url = url;
-	this.state = State.READY;
-	this.changeState();
+        this.url = url;
+        changeState(ProgressState.READY);
     }
 
     private HttpURLConnection openConnection(URL remoteFile) throws IOException {
-	return (HttpURLConnection) remoteFile.openConnection();
+        return (HttpURLConnection) remoteFile.openConnection();
     }
 
     private void saveConnectionToStream(HttpURLConnection connection, OutputStream outputStream)
-	    throws DownloadException {
-	int fileSize = connection.getContentLength();
-
-	try {
-	    BufferedInputStream inputStream = new BufferedInputStream(connection.getInputStream());
-	    BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(outputStream, BLOCK_SIZE);
-
-	    byte[] data = new byte[BLOCK_SIZE];
-	    int i;
-	    float totalDataRead = 0.0F;
-	    while ((i = inputStream.read(data, 0, BLOCK_SIZE)) >= 0) {
-		totalDataRead += i;
-		bufferedOutputStream.write(data, 0, i);
-
-		this.percentage = totalDataRead * 100 / fileSize;
-		this.state = State.PROGRESSING;
-		this.changeState();
-
-		if (Thread.currentThread().isInterrupted()) {
-		    throw new InterruptedException("The download has been aborted");
-		}
-	    }
-
-	    inputStream.close();
-	    bufferedOutputStream.close();
-	} catch (IOException | InterruptedException e) {
-	    this.state = State.FAILED;
-	    this.changeState();
-	    throw new DownloadException(String.format(EXCEPTION_ITEM_DOWNLOAD_FAILED, this.url), e);
-	}
-	this.state = State.SUCCESS;
-	this.changeState();
-
+            throws DownloadException {
+        int fileSize = connection.getContentLength();
+
+        try (BufferedInputStream inputStream = new BufferedInputStream(connection.getInputStream());
+                BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(outputStream, BLOCK_SIZE)) {
+            byte[] data = new byte[BLOCK_SIZE];
+            int i;
+            float totalDataRead = 0.0F;
+            while ((i = inputStream.read(data, 0, BLOCK_SIZE)) >= 0) {
+                totalDataRead += i;
+                bufferedOutputStream.write(data, 0, i);
+
+                this.percentage = totalDataRead * 100 / fileSize;
+                changeState(ProgressState.PROGRESSING);
```

Comment:
I think percentage could also be passed to changeState, the same way you've done for the state